### ISSUE TITLE
allow exception for new-cap if it's a named es6 import

### ISSUE THREAD

**Issue by jamestalmage** [eslint/eslint#11256]
TL;DR - don't complain about capitalization of named imports I have no control over.

**What rule do you want to change?**
`new-cap`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
default behavior, with optional config

**Please provide some example code that this change will affect:**

Some React HOC's define capitalized named exports that can't be used with new. Avoiding the error requires renaming the import.

```js
import {DragSource} from 'react-dnd';

const MyComponent = (props) => {
};

export default DragSource(MyComponent);
```

**What does the rule currently do for this code?**
Throws and error.

**What will the rule do after it's changed?**
Allow an exception when the identifier is a named import.

**Are you willing to submit a pull request to implement this change?**
yes.
